### PR TITLE
TCP2 endpoint refactoring

### PR DESCRIPTION
**Pull request by jukkar**
Removed dynamic endpoint allocation and place the needed endpoint struct in tcp connection struct. This avoid issues with memory allocation if we run out of memory.
Also noticed that the local address was incorrectly accessed, this was seen when log level was set to debug.

### CODE REVIEW

File: subsys/net/ip/tcp2.c [hunk 1]
```diff
@@ -1324,7 +1327,7 @@ static void tp_init(struct tcp *conn, struct tp *tp)
 	struct tp out = {
 		.msg = "",
 		.status = "",
-		.state = tcp_state_to_str(conn->state, true),
+		.state = tcp_state_to_str(conn->state),
```
Comment (ozhuraki) [inline subsys/net/ip/tcp2.c:1325]: There's a problem with this on TTCN-3 side, it needs the state with the prefix.

So, if you don't mind, let's leave tcp_state_to_str() as is.

**Comment by ozhuraki** [comment]
@jukkar 

I started to check and there's a problem to compile samples/net/sockets/tcp sample.

Could you please check?

File: subsys/net/ip/tcp2.c (hunk 1 above)
Comment (jukkar) [reply subsys/net/ip/tcp2.c:1325]: We need to figure out something, the logging functions do not work properly with static variables. I was seeing weird debug prints.

Comment (jukkar) [reply subsys/net/ip/tcp2.c:1325]: Lets drop that patch temporarily in order to continue with ttcn-3. I will send a new version.

**Comment by jukkar** [comment]
> I started to check and there's a problem to compile samples/net/sockets/tcp sample.
> 
> Could you please check?

Fixed

File: subsys/net/ip/tcp2.c [hunk 2]
```diff
@@ -756,6 +751,7 @@ static struct tcp *tcp_conn_new(struct net_pkt *pkt)
 	struct tcp *conn = NULL;
 	struct net_context *context = NULL;
 	sa_family_t af = net_pkt_family(pkt);
+	struct sockaddr local_addr = { 0 };
```
Comment (ozhuraki) [inline subsys/net/ip/tcp2.c:754]: How about to simplify this patch this way?

```
struct sockaddr *local_addr = net_sin_ptr(&context->local);
```

Comment (jukkar) [reply subsys/net/ip/tcp2.c:754]: sockaddr_in is not compatible with sockaddr_in_ptr (the address is pointer in latter in order to save some memory) so we need to copy

Comment (ozhuraki) [reply subsys/net/ip/tcp2.c:754]: What I meant, we can just get this pointer with net_sin_ptr() and use it, not?

**Review by ozhuraki — APPROVED** [review]
LGTM

For your judgement: I would still reduce/minimize the work around the local_addr in the second patch/commit, might be simpler/cleaner to just get it and work with it.

**Comment by jukkar** [comment]
> I would still reduce/minimize the work around the local_addr in the second patch/commit, might be simpler/cleaner to just get it and work with it.

Not trivial to get rid of local_addr, so I will postpone any effort for that. I am trying to get into data retransmission work but so far I have sidetracked too much :)

**Pull request merged**